src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheck.java

```diff
@@ -34,31 +40,74 @@
  * name but different signatures where the signature can differ by the number of
  * input parameters or type of input parameters or both.
  * </p>
+ * <ul>
+ * <li>
+ * Property {@code modifierGroups} - custom overload groups,
+ * allows for the definitions of multiple overloading method groups.
+ * Type is {@code java.util.regex.Pattern[]}.
+ * Default value is {@code ""}.
+ * </li>
+ * </ul>
  * <p>
  * To configure the check:
  * </p>
  * <pre>
- * &lt;module name=&quot;OverloadMethodsDeclarationOrder&quot;/&gt;
+ * &lt;module name="OverloadMethodsDeclarationOrder"/&gt;
+ * </pre>
+ * <p>
+ * Example of incorrect grouping of overloaded methods:
+ * </p>
+ * <pre>
+ * public void foo(int i) {} // OK
+ * public void foo(String s) {} // OK
+ * public void notFoo() {} // Violation, should be after 'foo(int i, String s)'
+ * public void foo(String s, int i) {} // OK
```

Comment:
Example is not correct from my testing. Violation goes on the next line not on `notFoo`. Examples should mimic behavior of code otherwise people will think behavior is not right as example says something else.